packages/lesmis-server/src/routes/PDFExportRoute.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ *
+ */
+import { Request, Response, NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import puppeteer from 'puppeteer';
+import Cookies from 'cookies';
+
+type SessionCookies = {
+  sessionCookieName: string;
+  sessionCookieValue: string;
+};
+
+type Body = {
+  pdfPageUrl: string;
+};
+
+export type PDFExportRequest = Request<
+  Record<string, never>,
+  { contents: Buffer },
+  Body,
+  Record<string, any>
+>;
+
+export class PDFExportRoute extends Route<PDFExportRequest> {
+  public constructor(req: PDFExportRequest, res: Response, next: NextFunction) {
+    super(req, res, next);
+    this.type = 'download';
+  }
+
+  private extractSessionCookie = (): SessionCookies => {
+    const cookies = new Cookies(this.req, this.res);
+    const sessionCookieName = 'sessionCookie';
+    const sessionCookieValue = cookies.get(sessionCookieName);
+    if (!sessionCookieValue) {
+      throw new Error(`'sessionCookie' is not found`);
+    }
+    return { sessionCookieName, sessionCookieValue };
+  };
+
+  private verifyBody = (body: any): Body => {
```

Comment:
Now that this.req is typed, you don't need to re-specify the type of the `body` arg, typescript knows it's a Body. And I don't think that return type is right? But anyway, you don't need to re-specify that either, typescript knows it's a string.

Reply:
Yeah type thing is one reason, another reason is to validate the schema of the body, do you think it is a bit unnecessary?

Reply:
I see, yeah, makes sense.